File: billion-loops/Benchmark.cs
using System;

class Benchmark
{
    static void Main()
    {
        var start = DateTime.Now;
        long sum = 0;
        long product = 1;

        for (long i = 0; i < 1_000_000_000; i++)
        {
            if (i % 2 == 0)
            {
                sum += i;
            }
            else
            {
                product = (product * (i % 100)) % 1_000_000_007;
            }

            if (i % 5 == 0)
            {
                sum -= i / 2;
            }
        }

        var elapsed = DateTime.Now - start;
        Console.WriteLine($"Sum: {sum}, Product: {product}, Elapsed Time: {elapsed.TotalSeconds} seconds");
    }
}

File: billion-loops/Benchmark.java
public class Benchmark {
    public static void main(String[] args) {
        long start = System.nanoTime();
        long sum = 0;
        long product = 1;
        long limit = 1_000_000_000L;

        for (long i = 0; i < limit; i++) {
            if (i % 2 == 0) {
                sum += i;
            } else {
                product = (product * (i % 100)) % 1_000_000_007L;
            }

            if (i % 5 == 0) {
                sum -= i / 2;
            }
        }

        long end = System.nanoTime();
        System.out.println("Final Sum: " + sum);
        System.out.println("Final Product: " + product);
        System.out.println("Elapsed Time: " + (end - start) / 1_000_000 + " ms");
    }
}
